ext/winevt/winevt_utils.cpp

```diff
@@ -500,3 +500,184 @@ get_description(EVT_HANDLE handle)
 
   return _wcsdup(result.data());
 }
+
+VALUE
+render_system_event(EVT_HANDLE hEvent)
+{
+  DWORD status = ERROR_SUCCESS;
+  EVT_HANDLE hContext = NULL;
+  DWORD dwBufferSize = 0;
+  DWORD dwBufferUsed = 0;
+  DWORD dwPropertyCount = 0;
+  VALUE vRenderedValues;
+  PEVT_VARIANT pRenderedValues = NULL;
+  WCHAR wsGuid[50];
+  LPSTR pwsSid = NULL;
+  ULONGLONG ullTimeStamp = 0;
+  ULONGLONG ullNanoseconds = 0;
+  SYSTEMTIME st;
+  FILETIME ft;
+  CHAR buffer[32];
+  VALUE rbstr;
+  DWORD EventID;
+  VALUE hash = rb_hash_new();
+
+  hContext = EvtCreateRenderContext(0, NULL, EvtRenderContextSystem);
+  if (NULL == hContext) {
+    rb_raise(rb_eWinevtQueryError,
+             "Failed to create renderContext with %lu\n", GetLastError());
+  }
+
+  if (!EvtRender(hContext,
+                 hEvent,
+                 EvtRenderEventValues,
+                 dwBufferSize,
+                 pRenderedValues,
+                 &dwBufferUsed,
+                 &dwPropertyCount)) {
+    if (ERROR_INSUFFICIENT_BUFFER == (status = GetLastError())) {
+      dwBufferSize = dwBufferUsed;
+      pRenderedValues = (PEVT_VARIANT)ALLOCV(vRenderedValues, dwBufferSize);
+      if (pRenderedValues) {
+        EvtRender(hContext,
+                  hEvent,
+                  EvtRenderEventValues,
+                  dwBufferSize,
+                  pRenderedValues,
+                  &dwBufferUsed,
+                  &dwPropertyCount);
+      } else {
+        EvtClose(hContext);
+        rb_raise(rb_eRuntimeError,
+             "Failed to malloc memory with %lu\n", status);
+      }
+    }
+
+    if (ERROR_SUCCESS != (status = GetLastError())) {
+      EvtClose(hContext);
+      ALLOCV_END(vRenderedValues);
+
+      rb_raise(rb_eWinevtQueryError,
+               "EvtRender failed with %lu\n", status);
+    }
+  }
+
+  rbstr = wstr_to_rb_str(CP_UTF8, pRenderedValues[EvtSystemProviderName].StringVal, -1);
```

Comment:
Please consider adding comments to point API reference for these values.
Otherwise it's hard to confirm that they are correct or not.
I found following documents. Are they the same as the document which you referred to?

* https://docs.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtrender
* https://docs.microsoft.com/en-us/windows/win32/api/winevt/ns-winevt-evt_variant
* https://docs.microsoft.com/en-us/windows/win32/api/winevt/ne-winevt-evt_variant_type
* https://msdn.microsoft.com/en-us/windows/aa385606(v=vs.80)?f=255&MSPPError=-2147217396

Reply:
I've confirmed all members.
LGTM